### DISCUSSION TITLE
Adding charged species to training set

### DISCUSSION THREAD

**Discussion by b-rousse** [ACEsuit/mace#746] [Q&A] [answered]
Hi all, I'd appreciate your expertise on the following:

I am aware that MACE does not explicitly consider long-range charge effects but that short-range charge effects can be modeled quite well with MACE. I am also aware that retraining MACE is performed on absolute energies and forces, making it imperative that any new data be provided with the same zero of energy.

I would like to retrain MACE on two dissociation curves: one of a cationic molecule dissociating from a neutral molecule, and one of a different molecule in anionic form dissociating from a neutral molecule.

I _think_ it should be okay to include the dissociation curve with charged species into the training data alongside the SPICE dataset, as we could have an equivalent system where the PFAS was protonated, and then the proton is removed to a distance of >10 Å (i.e. beyond the interaction length of MACE). As the energy of the isolated proton is zero within the Born-Oppenheimer approximation, I don't believe its exclusion will change the zero of energy that is important to preserve for a consistent training set. Is this assessment correct?

Also worth noting that the training calculations will be done in gas phase, I suppose I could use CPCM if the charge screening is an issue.

Thank you!

**Comment by gabor1** [comment, marked as answer]
The problem is about inconsistent data where the geometries are very close but the charge state (and hence the energy) is very different. So an isolated proton has very different energy from the isolated H, but geometrically the same. That's the problem. Electrostatic mace will solve it in a few months. In the meantime, you could try condensed phase training with some charged species, because there might be sufficient geometric information there to learn the energies (and implicitly the charge state). But I wouldn't mess with the isolated proton, that's really inconsistent.

**Reply by gabor1** [reply]
If you really want to experiment, you can retrain (but not finetune!) from scratch taking samples from SPICE and your own data, but give the isolated atom energies from calculations where you ONLY include a proton (i.e. E=0) rather than H (E=-13.6). that's at least consistent. Just don't ever add H2 to your dataset :-)

**Reply by b-rousse** [reply]
I see, thank you for the detailed response. I'll try with explicit solvent and eagerly await electrostatic MACE!

**Reply by husakm** [reply]
Any anouncment when MACE with electrostatic support will be available ? I have in mind to mix MACE OFF23/24 with external elecrostatic correction, but I am not sure it have sense.

**Reply by gabor1** [reply]
soon. if you are very impatient, there is a PR for a simple geometry-dependent charge scheme, you can try [this](https://github.com/ACEsuit/mace/pull/934)

**Reply by husakm** [reply]
Can I have a few question:
1) Will it be required to supply for MACE with electrostatic partial charges on each atom ? What sort of them - RESP charges fitted to QM or some sort of charges compatible with the training set ?
2) Will it work with periodic structures (molecular crystals) when electrostatic works on long range and its contribution is not straight-forward to calculate (Ewald sumation) ?

Why I ask: I was testing MACE for crystal structure prediction of molecular crystals. Non ionic - perfect results. Salts - total failure.

**Reply by gabor1** [reply]
1. For the PR mentioned above, you don't need to supply anything. For the upcoming electrostatic MACEs from us, there will be two versions, in one you again don't specify anything new as input. For the other, you specify oxidation states in the form of integer charges on each atom, which will be added to by the ML prediction to arrive at partial charges. 
2. yes, all of our packages work for periodic systems

**Comment by husakm** [comment]
Understand. We want  to use MACE for complete CSD crystal structures database processing(validation of the experimental structures) The integer charges on individual atoms are not directly given. Automatic assignment of formal charges to atoms is far from trivial, see:
https://github.com/rdkit/rdkit/discussions/8021  
So I wander how does first method without formal charges work ...
Looking forward to the new version.

**Reply by gabor1** [reply]
for this sort of application, I would take a separate partial charge model, subtract its energies and forces, and fit the remainder with mace.

**Reply by husakm** [reply]
You mean:
1) calculate the energy by QM, chose a charge model, substract the electrostatic from the learning energy
2) learn MCE on the energies without electrostatic
3) for final energy and geometry calculation combine MACE and identical charge model as substraced
The same can be eventualy done for long range dispersion correction.
I agree, but it is to much work for me ... Any chance somebody else will do this ?

**Reply by gabor1** [reply]
I think for some kind of validation, you can just use the existing mace-off (not for salts, but crystals with neutral molecules). for accurate prediction of crystal structure formation energies, you indeed need an electrostatic long range model, either one built into the model, or an external one, along the lines of the steps you say above. As I mentioned above, there is now a PR which allows very simple charge prediction built into the mace model, and you don't need to precompute or subtract anything, but you need to train a new model.

**Reply by husakm** [reply]
MACE OFF23 was already succesfuly tested for crystal structure prediction (ranking of suggestions).
https://pubs.rsc.org/en/content/articlelanding/2025/cp/d5cp00593k
The results are comparable to DFT at fraction of cost ...
It fails totaly for salts ...
I am not an expert in ML so I should wait until somebody more skiled will generate a model with elctrostatic support ...

**Reply by gabor1** [reply]
indeed. we are working on an electrostatic version ourselves.
